### PR TITLE
add S3 lifecycles support

### LINKED ISSUE #169: S3 - Support bucket lifecycle configuration
Currently the S3 Handel service doesn't let you configure any bucket lifecycles. We should allow this feature (including potentially the built-in integration with Glacier).

### PR DESCRIPTION
**Pull request by ndpete**
This add initial support for s3 bucket lifecycle policies

resolves #169

### CODE REVIEW

File: docs/supported-services/s3.rst [hunk 1]
```diff
@@ -54,12 +53,80 @@ This service takes the following parameters:
...
+     - Lifecycle Policies to apply to the bucket. See `AWS Docs for more info <http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-lifecycleconfig-rule.html>`_
    * - tags
      - :ref:`s3-tags`
      - No
      -
      - Any tags you want to apply to your S3 bucket
 
+.. _s3-lifecycles:
+
+lifecycles
```
Comment (dsw88) [inline docs/supported-services/s3.rst]: I usually try to capitalize these section headers, so "Lifecycles"

File: docs/supported-services/s3.rst [hunk 2]
```diff
@@ -54,12 +53,80 @@ This service takes the following parameters:
...
+~~~~~~~~~~
+A list of life cycle rules
```
Comment (dsw88) [inline docs/supported-services/s3.rst]: It looks like you're missing docs for the "prefix" field. Is that on the "lifecycles" section or under each "transitions" section?

Comment (ndpete) [inline docs/supported-services/s3.rst]: Oh good catch, thats on the Lifecycles section right under name before transitions most likely.

Comment (ndpete) [inline docs/supported-services/s3.rst]: I'll clean those up if you'd like

**Pull request merged**